src/loadDefinedFile.js

```diff
@@ -91,8 +91,12 @@ function tryYaml(content: string, filepath: string, cb: () => ?Object) {
 
 function tryRequire(content: string, filepath: string, cb: () => ?Object) {
   try {
-    return requireFromString(content, filepath);
+    return requireJs(content, filepath);
   } catch (e) {
+    if (e.message.indexOf('must use default export with ES Modules') !== -1) {
+      throw e;
+    }
+
```

Comment:
@olsonpm I think this is expected/wanted behavior here. "[throws an error if non-default es module with missing extension](https://github.com/davidtheclark/cosmiconfig/pull/113/files#diff-06c2c9f7946f98248f0f685ee7c0960eR245)" is the respective test. Otherwise "Failed to parse" error is thrown. Please confirm.

Reply:
that would be my expectation.  However support for "precompiled modules" as a whole may not be what sudo-suhas and davidtheclark have in mind - so it'd be best to wait for them before figuring out specifics.

Reply:
https://github.com/davidtheclark/cosmiconfig/issues/109#issuecomment-372137187 leaves me with the impression that is what is wanted. I definitely could be misunderstanding though.

Thanks for the help with this PR! I appreciate it.

Reply:
Ah - yeah my interpretation was not to worry about looking for named exports for a configuration.  Your interpretation makes more sense though :)